### PR TITLE
Edit and delete functionality added and controlled with interaction with the swagger backend api

### PR DESCRIPTION
**Pull request by Moraheem98**
- created separate thunk functions for adding, editing and deleting functionality.
- Also cleaned up some additional logic in the forms components.

### CODE REVIEW

File: task-2/src/components/MovieCard/index.js [hunk 1]
```diff
@@ -3,31 +3,45 @@ import { useDispatch } from 'react-redux';
...
+import { deleteMovie } from '../../store/thunk';
+import { switchSelectedMovie } from '../../store/actionCreators';
 
 import './index.css';
 
-export const MovieCard = ({ title, genres, release_date, show, onClick }) => {
+export const MovieCard = ({ movie, onClick, setOpenModal }) => {
 	const dispatch = useDispatch();
+
+	const onClickFunction = (event) => {
```
Comment (Dmitry-White) [inline task-2/src/components/MovieCard/index.js]: Consider giving this handler a more meaningful name.

File: task-2/src/components/Formik/movieValidationSchema.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ export const movieValidationSchema = Yup.object().shape({
 	title: Yup.string().required('Title is required'),
 	release_date: Yup.string().required('Release Date is required'),
 	poster_path: Yup.string().required('Image Url is required'),
-	genres: Yup.string().required('Genre is required'),
+	// genres: Yup.string().required('Genre is required'),
```
Comment (Dmitry-White) [inline task-2/src/components/Formik/movieValidationSchema.js]: Consider removing commented pieces of code.

File: task-2/src/core/api/methods.js [hunk 3]
```diff
@@ -31,3 +31,34 @@ export const requestMoviesByRating = async () => {
...
+	await restRequest(movie, 'POST', moviesEndpoint);
+};
+
+export const editMovieRequest = async (movie) => {
+	const { moviesEndpoint } = API_CONFIG;
+	await restRequest(movie, 'PUT', moviesEndpoint);
+};
+export const deleteMovieRequest = async (movie) => {
+	const { moviesEndpoint } = API_CONFIG;
+	const string = `${moviesEndpoint}/${movie.id}`;
```
Comment (Dmitry-White) [inline task-2/src/core/api/methods.js]: Consider avoiding using name that could shadow reserved ones.
"url" is a better name for this variable.

File: task-2/src/store/reducers/moviesReducer.js [hunk 4]
```diff
@@ -1,4 +1,9 @@
-import { SET_MOVIES, ADD_MOVIE, DELETE_MOVIE } from '../actionTypes';
+import {
+	SET_MOVIES,
+	// ADD_MOVIE,
```
Comment (Dmitry-White) [inline task-2/src/store/reducers/moviesReducer.js]: Consider going through the codebase and either delete commented code or leave a TODO comment explaining why it's still there.

**Review by Dmitry-White — CHANGES_REQUESTED** [review]
Really well done!

File: task-2/src/components/MovieCard/index.js (hunk 1 above)
Comment (Moraheem98) [reply task-2/src/components/MovieCard/index.js]: Resolved.

File: task-2/src/components/Formik/movieValidationSchema.js (hunk 2 above)
Comment (Moraheem98) [reply task-2/src/components/Formik/movieValidationSchema.js]: Resolved.

**Review by Dmitry-White — APPROVED** [review]
Approving to be merged after the above comments are addressed

File: task-2/src/core/api/methods.js (hunk 3 above)
Comment (Moraheem98) [reply task-2/src/core/api/methods.js]: resolved.

File: task-2/src/store/reducers/moviesReducer.js (hunk 4 above)
Comment (Moraheem98) [reply task-2/src/store/reducers/moviesReducer.js]: resolved.

**Pull request merged**